assets/js/googlesitekit/data/create-fetch-store.js

```diff
@@ -157,7 +157,8 @@ export const createFetchStore = ( {
 			type: START_FETCH,
 		};
 
-		yield clearError( baseName, args );
+		// @TODO: remove clearMatchingLegacyError usage once all instances of the legacy behavior have been removed.
+		yield clearError( baseName, args, { clearMatchingLegacyError: true } );
```

Comment:
The `clearMatchingLegacyError` param is used here in order to address a logical error, where further down this function, when an error is received it's set in the store twice (once in `state.errors` keyed by the error key, and also in the single `state.error` property). However in the prelude here only the keyed error in `state.errors` is removed. This means that a stale fetch error can remain in state following a refetch.

This became evident when making the change to `ErrorNotice`, when successfully retrying a selector that results in a fetch, the stale error was still displayed.

Reply:
From my perspective this isn't too bad, is well-tested, and is clearly documented.

I think this is a fine workaround for a temporary state of things 👍🏻

Reply:
Thanks for the feedback @tofumatt!